Content/Items/Magnet/Weapons.GrayGoo.cs

```diff
@@ -0,0 +1,435 @@
+			Item.shoot = ProjectileType<GrayGooProj>();
+			Item.knockBack = 0;
+			Item.noUseGraphic = true;
+		}
+
+		public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+		{
+			// This is needed so the buff that keeps your minion alive and allows you to despawn it properly applies
+			player.AddBuff(Item.buffType, 2);
+
+			// Here you can change where the minion is spawned. Most vanilla minions spawn at the cursor position.
+			position = Main.MouseWorld;
+
+			var proj = Projectile.NewProjectileDirect(source, position, velocity, type, damage, knockback, player.whoAmI);
+			proj.originalDamage = damage;
+			return false;
+		}
+	}
+
+	public class GrayGooProj : ModProjectile
+	{
+		public const int maxMinionChaseRange = 2000;
+
+		public float lerper;
+
+		public float oldLerper;
+
+		public static RenderTarget2D NPCTarget;
+
+		public bool foundTarget;
+
+		public float oldEnemyWhoAmI;
+
+		public ref float EnemyWhoAmI => ref Projectile.ai[1];
+
+		public Player Owner => Main.player[Projectile.owner];
+
+		public override string Texture => AssetDirectory.Invisible;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Gray Goo");
+
+			Main.projPet[Projectile.type] = true; // Denotes that this Projectile is a pet or minion
+			ProjectileID.Sets.MinionTargettingFeature[Projectile.type] = true; // This is necessary for right-click targeting
+			ProjectileID.Sets.MinionSacrificable[Projectile.type] = true; // This is needed so your minion can properly spawn when summoned and replaced when other minions are summoned			 
+			ProjectileID.Sets.CultistIsResistantTo[Projectile.type] = true; // Don't mistake this with "if this is true, then it will automatically home". It is just for damage reduction for certain NPCs
+		}
+
+		public override void Load()
+		{
+			if (Main.dedServ)
+				return;
+
+			ResizeTarget();
+
+			On.Terraria.Main.CheckMonoliths += DrawNPCtarget;
+		}
+
+		public static void ResizeTarget()
+		{
+			Main.QueueMainThreadAction(() => NPCTarget = new RenderTarget2D(Main.graphics.GraphicsDevice, Main.screenWidth, Main.screenHeight));
+		}
+
+		public override void SetDefaults()
+		{
+			Projectile.width = 16;
+			Projectile.height = 16;
+
+			Projectile.tileCollide = false;
+
+			Projectile.minion = true;
+			Projectile.minionSlots = 1;
+			Projectile.penetrate = -1;
+			Projectile.timeLeft = 5000;
+			Projectile.friendly = true;
+			Projectile.usesLocalNPCImmunity = true;
+			Projectile.localNPCHitCooldown = 15;
+		}
+
+		public override bool? CanCutTiles()
+		{
+			return false;
+		}
+
+		public override bool MinionContactDamage()
+		{
+			return true;
+		}
+
+		public override bool? CanHitNPC(NPC target)
+		{
+			if (target.whoAmI != (int)EnemyWhoAmI)
+				return false;
+			return base.CanHitNPC(target);
+		}
+
+		public override void AI()
+		{
+			#region Active check
```

Comment:
This region should not be here, it serves no purpose and adds needless verbosity

Reply:
fixed